Add set/getChatboxCharacterLimit and increase max chat message length to 255

Lpsd opened the pull request:
This PR resolves #2090 by adding set/getChatboxCharacterLimit functions to the **client-side only.**

It also increases the max chat message length to 255 (not including player name or "(TEAM)" tags), over the previous maximum (and default) of 96. The default value will still stay at 96.

___

### setChatboxCharacterLimit
```lua
bool setChatboxCharacterLimit(int charLimit)
```
Passing -1 will reset the character limit  back to 96.
If you pass anything other than an integer, or outside the range of 0-255 (excluding -1), it will return false.
      
___

### getChatboxCharacterLimit
```lua
int getChatboxCharacterLimit()
```
Pretty self-explanatory.

___

When a player disconnects from a server, their character limit is set back to the default (96).

Lpsd committed:
Initial commit

Lpsd committed:
Add max input char limit and max input line values as const members to CChat

Lpsd committed:
Add GetChatboxMaxCharacterLimit to CCore (for CChat)

Lpsd committed:
Update CCoreInterface virtual method ordering

Lpsd committed:
Increase MAX_CHAT_LENGTH and make MAX_CHATECHO_LENGTH "dynamic"

Lpsd committed:
Fix Packet_ChatEcho to use MAX_CHATECHO_LENGTH instead of MAX_OUTPUTCHATBOX_LENGTH

Why was it using MAX_OUTPUTCHATBOX_LENGTH all this time? lmao

Lpsd committed:
Make setChatboxCharacterLimit reset on passing -1, instead of none

Otherwise you could pass anything other than a number to setChatBoxCharacterLimit, and you wouldn't know anything was wrong, yet the chatbox input limit would reset

Lpsd committed:
Remove extra func definition added by accident

Lpsd commented:
On hold, waiting for my review / merge of #2170 

Will need to address conflicts in this PR after that.

Pirulax commented on Client/sdk/core/CCoreInterface.h:
```diff
@@ -180,6 +176,11 @@ class CCoreInterface
+    virtual bool SetChatboxCharacterLimit(int charLimit) = 0;
+    virtual void ResetChatboxCharacterLimit() = 0;
+    virtual int  GetChatboxCharacterLimit() = 0;
+    virtual int  GetChatboxMaxCharacterLimit() = 0;
```
For further reference: Netcode (presumeably) uses CCoreInterface. If the order is changed it obviously crashes. This shouldn't be an issue in production nightlies though.

patrikjuvonen commented on Client/mods/deathmatch/logic/luadefs/CLuaGUIDefs.cpp:
```diff
@@ -281,6 +287,7 @@ void CLuaGUIDefs::AddGuiElementClass(lua_State* luaVM)
     lua_classvariable(luaVM, "text", "guiSetText", "guiGetText");
     lua_classvariable(luaVM, "size", "guiSetSize", "guiGetSize");
     lua_classvariable(luaVM, "position", "guiSetPosition", "guiGetPosition");
+    //lua_classvariable(luaVM, "chatboxCharacterLimit", "setChatboxCharacterLimit", "getChatboxCharacterLimit");
```
Why is this commented out?

Lpsd replied:
no good reason that I can think of, probably just forgot to uncomment

patrikjuvonen commented on Client/mods/deathmatch/logic/Config.h:
```diff
@@ -52,17 +52,17 @@ class CVaPassNext
 /*** va_pass() (passing of ... variable length arguments ***/
 
-// Min and max number of characters in chat messages
+// Min and max number of characters in chat messages (the message itself, not including player nick)
```
I wish there was a way to merge these client and server configs together in a shared config file, but maybe out of scope for this PR.

Lpsd replied:
Yeah it also needs to be made clearer what exactly the difference between max chat length, and max **chat echo** length is (among other things). 

I've cleared it up a little bit here hopefully, including fixing an oversight in `Packet_ChatEcho` where `MAX_OUTPUTCHATBOX_LENGTH` was being used instead of `MAX_CHATECHO_LENGTH`.

patrikjuvonen committed:
Merge branch 'master' into chatbox-limit

Lpsd committed:
Merge branch 'master' into chatbox-limit

botder commented on Client/mods/deathmatch/logic/luadefs/CLuaGUIDefs.cpp:
```diff
@@ -281,6 +287,7 @@ void CLuaGUIDefs::AddGuiElementClass(lua_State* luaVM)
     lua_classvariable(luaVM, "text", "guiSetText", "guiGetText");
     lua_classvariable(luaVM, "size", "guiSetSize", "guiGetSize");
     lua_classvariable(luaVM, "position", "guiSetPosition", "guiGetPosition");
+    //lua_classvariable(luaVM, "chatboxCharacterLimit", "setChatboxCharacterLimit", "getChatboxCharacterLimit");
```
```suggestion
    lua_classvariable(luaVM, "chatboxCharacterLimit", "setChatboxCharacterLimit", "getChatboxCharacterLimit");
```

botder commented on Client/core/CChat.h:
```diff
@@ -203,6 +202,11 @@ class CChat
+    void      SetCharacterLimit(int charLimit);
+    int       GetCharacterLimit() const { return m_iCharacterLimit; }
+    const int GetDefaultCharacterLimit() const { return m_iDefaultCharacterLimit; }
+    const int GetMaxCharacterLimit() const { return m_iMaxCharacterLimit; }
```
```suggestion
    int GetDefaultCharacterLimit() const { return m_iDefaultCharacterLimit; }
    int GetMaxCharacterLimit() const { return m_iMaxCharacterLimit; }
```

botder commented on Client/core/CChat.h:
```diff
@@ -292,4 +296,9 @@ class CChat
+    int m_iCharacterLimit;
+    const int m_iDefaultCharacterLimit = 96;
+    const int m_iMaxCharacterLimit = 255;
+    const int m_iMaxInputLines = 5;
```
It would make more sense to use `static inline constexpr` constants instead.

botder committed:
Apply suggestions from code review

patrikjuvonen committed:
Merge branch 'master' into chatbox-limit

patrikjuvonen committed:
Merge branch 'master' into chatbox-limit

patrikjuvonen committed:
Apply constexpr suggestion from botder's code review

Suggestion at https://github.com/multitheftauto/mtasa-blue/pull/2215#discussion_r967852686

patrikjuvonen reviewed (approved):
Tested working, looks good.
